system/security/shared/caching.hpp

```diff
@@ -151,16 +151,25 @@ class CPermissionsCache
 
     virtual ~CPermissionsCache();
 
-    static CPermissionsCache* queryInstance()
+    //Returns a shared cache of a given class type.
+    //Call this method with a unique class string ("LDAP", "MyOtherSecMgr")
+    //to create a cache shared amongst security managers of the same class
+    static CPermissionsCache* queryInstance(const char * _secMgrClass)
     {
+        const char * secMgrClass = (_secMgrClass != nullptr  &&  *_secMgrClass) ? _secMgrClass : "genericSecMgrClass";
```

Comment:
Because this is only called at ESP startup, I don't think its worth it to have both the map and a static instance pointer to manage. FWIW I don't think anyone would ever call it with NULL